c10/util/ArrayRef.h

```diff
@@ -22,10 +22,87 @@
 
 #include <array>
 #include <iterator>
+#include <memory>
+#include <type_traits>
 #include <vector>
 
 namespace c10 {
 
+
+
+struct TORCH_API SymbolicIntImpl {
+  bool virtual isSymbolicInt(int64_t data_) = 0;
+
+  // I can't overload + for primitive types
+  int64_t virtual add(int64_t s1, int64_t s2) = 0;
+  bool virtual equal(int64_t s1, int64_t s2) = 0;
+};
+
+
+class TORCH_API SymbolicOrConcreteInt {
```

Comment:
Do we expect this to be used for any int?
I feel like the "trick" being used is very specific to size so maybe we want to simply call this `SizeVal(ue)`?

Reply:
I think it could be used for any symbolic ints, not just for sizes.